mayastor/src/grpc/mayastor_grpc.rs

```diff
@@ -421,6 +421,10 @@ impl mayastor_server::Mayastor for MayastorSvc {
             Ok(ListNexusReply {
                 nexus_list: instances()
                     .iter()
+                    .filter(|n| {
```

Comment:
Aren't we loosing observability here ? I mean, shouldn't MOAC filter out not fully initialised nexuses rather than Mayastor hide them ?

Reply:
I am in agreement. However, unfortunately the Nexus Init state gets mapped to the Degraded state before being passed up to MOAC (as MOAC only knows about 3 states; online, degraded and faulted) - https://github.com/openebs/Mayastor/blob/72a273707e8f2a1f09bcdf18de999873df2db89e/mayastor/src/bdev/nexus/nexus_bdev.rs#L941

As far as MOAC is concerned the Degraded state doesn't prevent the nexus from being published and used by an application. What we needed to do here was prevent the nexus from being able to be used before its persistent information had been written to the store - which occurs whilst in the Init state. After speaking with Jeffry, the suggestion was to prevent initialising nexuses from being reported back to MOAC.

Reply:
So we all agree here :)